source/main.c

```diff
@@ -183,11 +183,25 @@ Result loadCode()
     fclose(file);
     printf(".code written to %s\n", code_path);
     
-    if(tid_short == 0x00055E00 || tid_short == 0x00055D00 || tid_short == 0x0011C400 || tid_short == 0x0011C500)
+    if(tid_short == 0x00055E00 || tid_short == 0x00055D00 || tid_short == 0x0011C400 || tid_short == 0x0011C500) // || tid_short == 0x00164800 || tid_short == 0x00175E00)
     {
-        u32 ser_addr = search_string(tid_short == 0x00055E00 || tid_short == 0x00055D00 ? "https://3ds1-fushigi.pokemon-gl.com/api/" : "https://3ds2-fushigi.pokemon-gl.com/api/", tid_short == 0x0011C400 || tid_short == 0x0011C500 ? 0x4E0000 : 0x490000, 0x100000, paramblk->code_data);
-        u32 boss1_addr = search_string("https://npdl.cdn.nintendowifi.net/p01/nsa/%s/%s/%s", tid_short == 0x0011C400 || tid_short == 0x0011C500 ? 0x4E0000 : 0x490000, 0x100000, paramblk->code_data);
-        u32 boss2_addr = search_string("https://npfl.c.app.nintendowifi.net/p01/filelist/", tid_short == 0x0011C400 || tid_short == 0x0011C500 ? 0x4E0000 : 0x490000, 0x100000, paramblk->code_data);
+        u32 ser_addr = 0;
+        u32 boss1_addr = 0;
+        u32 boss2_addr = 0;
+        if(tid_short == 0x00055E00 || 0x00055D00) { // Y || X
+            ser_addr = search_string("https://3ds1-fushigi.pokemon-gl.com/api/", 0x490000, 0x100000, paramblk->code_data);
+            boss1_addr = search_string("https://npdl.cdn.nintendowifi.net/p01/nsa/%s/%s/%s", 0x490000, 0x100000, paramblk->code_data);
+            boss2_addr = search_string("https://npfl.c.app.nintendowifi.net/p01/filelist/", 0x490000, 0x100000, paramblk->code_data);
+        }
+        else if(tid_short == 0x0011C400 || 0x0011C500) { // OR || AS
+            ser_addr = search_string("https://3ds2-fushigi.pokemon-gl.com/api/", 0x4E0000, 0x100000, paramblk->code_data);
+            boss1_addr = search_string("https://npdl.cdn.nintendowifi.net/p01/nsa/%s/%s/%s", 0x4E0000, 0x100000, paramblk->code_data);
+            boss2_addr = search_string("https://npfl.c.app.nintendowifi.net/p01/filelist/", 0x4E0000, 0x100000, paramblk->code_data);
+        }
+        // else { // (tid_short == 0x00164800 || 0x00175E00) aka SUN || MOON
```

Comment:
if you want to use MysteryMachine-Patcher in hbl with sun moon
you also should use this
https://github.com/ihaveamac/9.6-dbgen-xmls